src/libraries/System.Net.NetworkInformation/src/System/Net/NetworkInformation/SystemTcpConnection.cs

```diff
@@ -11,7 +11,7 @@ internal class SystemTcpConnectionInformation : TcpConnectionInformation
         private readonly IPEndPoint _remoteEndPoint;
         private readonly TcpState _state;
 
-        internal SystemTcpConnectionInformation(Interop.IpHlpApi.MibTcpRow row)
+        internal SystemTcpConnectionInformation(in Interop.IpHlpApi.MibTcpRow row)
```

Comment:
We will initialize the connection information directly from the buffer that was filled in by the OS API, without multiple intermediate copies and allocations.